syntax = "proto3";

package message_service;

import "google/protobuf/timestamp.proto";

message Message {
    string message_id = 1;
    string sender_email = 2;
    string receiver_email = 3;
    string content = 4;
    google.protobuf.Timestamp timestamp = 5;
}

message SendMessageRequest {
    Message message = 1;
}

message SendMessageResponse {
    bool success = 1;
    string status_message = 2;  // Cambio de "message" para evitar confusiÃ³n
}

message GetMessagesRequest {
    string user_email = 1;
    int32 page = 2;
    int32 page_size = 3;
}

message GetMessagesResponse {
    repeated Message messages = 1;
}

// Servicio Message Service
service MessageService {
    rpc SendMessage(SendMessageRequest) returns (SendMessageResponse);
    rpc GetMessages(GetMessagesRequest) returns (GetMessagesResponse);
}
